packages/plugin-ma-tracking/src/ma-tracking.ts

```diff
@@ -0,0 +1,89 @@
+import {
+  AdditionalBrowserOptions,
+  BaseEvent,
+  BeforePlugin,
+  BrowserConfig,
+  Campaign,
+  Event,
+  PluginType,
+  Storage,
+} from '@amplitude/analytics-types';
+import type { AmplitudeBrowser } from '@amplitude/analytics-browser';
+import { AdvancedCampaignTracker } from './advanced-campaign-tracker';
+
+export class MATrackingPlugin implements BeforePlugin {
+  name = 'MATracking';
+  type = PluginType.BEFORE as const;
+
+  constructor(private readonly instance: AmplitudeBrowser, private readonly options: AdditionalBrowserOptions = {}) {}
+
+  async setup(config: BrowserConfig): Promise<undefined> {
+    const attributionConfig = this.options.attribution ?? {};
+    const pageTrackingConfig =
+      this.options.trackPageViews ??
+      (attributionConfig.trackPageViews && {
+        filter: 'onAttribution',
+      });
+
+    const instance = this.instance;
+    // Share cookie storage with user session storage
+    const storage = config.cookieStorage as unknown as Storage<Campaign>;
```

Comment:
As unknown?

Reply:
cookieStorage is `Storage<UserSession>`. In order to support `Storage<Campaign>`, need to set `as unknown` first.

Reply:
again, this bypasses type checking. defeats the purpose of using typescript

Reply:
i think we can use `createFlexibleStorage` here instead of reusing cookie storage

Reply:
`createFlexibleStorage` is not a shared component. We update the plugin when it is moved into common package.